src/MICore/Debugger.cs

```diff
@@ -1161,15 +1136,14 @@ private void HandleThreadGroupExited(Results results)
                 {
                     if (_debuggeePids.Remove(threadGroupId))
                     {
-                        if (_debuggeePids.Count == 0)
-                            ProcessState = ProcessState.Exited;
+                        isThreadGroupEmpty = _debuggeePids.Count == 0;
                     }
                 }
             }
 
-            if (ProcessState == ProcessState.Exited)
+            if (isThreadGroupEmpty)
             {
-                OnDebuggerProcessExit(null);
+                ScheduleStdOutProcessing(@"*stopped,reason=""exited""");
```

Comment:
Tested scenarios:
1. Attach to a process, set a BP, run debuggee to the end, make sure the terminal window opened by VSCode is closed (bug 197013)
2. Launch VSCode, start debugging, then close VSCode, make sure the terminal window opened by VSCode closed (Bug 196499)
3. Run as root, launch VSCode, start debugging, then stop debugging, make sure the terminal window closed. (Bug 194992)